lib/utf8/index.js

```diff
@@ -38,36 +38,27 @@ utf8.length = function utf8_length(string) {
  * @returns {string} String read
  */
 utf8.read = function utf8_read(buffer, start, end) {
-    var len = end - start;
-    if (len < 1)
+    if (end - start < 1) {
         return "";
-    var parts = null,
-        chunk = [],
-        i = 0, // char offset
-        t;     // temporary
-    while (start < end) {
-        t = buffer[start++];
-        if (t < 128)
-            chunk[i++] = t;
-        else if (t > 191 && t < 224)
-            chunk[i++] = (t & 31) << 6 | buffer[start++] & 63;
-        else if (t > 239 && t < 365) {
```

Comment:
My understanding of this function is that the passed in buffer is always an array of uint8s, each byte being a utf8 code unit. The condition `t < 365` is a bit odd under this assumption, from profiling it actually forces v8 to deopt when `t` is compared to a non uint8 value. Alternatively I may have grossly misunderstood how this is expected to work.